core/src/main.cpp

```diff
@@ -0,0 +1,865 @@
+		client.at("port").get<int>(),
+		irods::experimental::fully_qualified_username{username, zone},
+		irods::experimental::fully_qualified_username{username, zone},
+		[pw = rodsadmin.at("password").get<std::string>()](RcComm& _comm) mutable {
+			if (const auto ec = clientLoginWithPassword(&_comm, pw.data()); ec != 0) {
+				throw std::invalid_argument{fmt::format("Could not authenticate rodsadmin user: [{}]", ec)};
+			}
+		},
+		opts);
+} // init_irods_connection_pool
+
+class process_stash_eviction_manager
+{
+	net::steady_timer timer_;
+	std::chrono::seconds interval_;
+
+  public:
+	process_stash_eviction_manager(net::io_context& _io, std::chrono::seconds _eviction_check_interval)
+		: timer_{_io}
+		, interval_{_eviction_check_interval}
+	{
+		evict();
+	} // constructor
+
+  private:
+	auto evict() -> void
+	{
+		timer_.expires_after(interval_);
+		timer_.async_wait([this](const auto& _ec) {
+			if (_ec) {
+				return;
+			}
+
+			log::trace("Evicting expired items ...");
+			irods::http::process_stash::erase_if([](const auto& _k, const auto& _v) {
+				const auto* client_info = boost::any_cast<const irods::http::authenticated_client_info>(&_v);
+				const auto erase_value = client_info && std::chrono::steady_clock::now() >= client_info->expires_at;
+
+				if (erase_value) {
+					log::debug("Evicted bearer token [{}].", _k);
+				}
+
+				return erase_value;
+			});
+
+			evict();
+		});
+	} // evict
+}; // class process_stash_eviction_manager
+
+auto main(int _argc, char* _argv[]) -> int
+{
+	po::options_description opts_desc{""};
+
+	// clang-format off
+	opts_desc.add_options()
+		("config-file,f", po::value<std::string>(), "")
+		("jsonschema-file", po::value<std::string>(), "")
+		("dump-config-template", "")
+		("dump-default-jsonschema", "")
+		("help,h", "")
+		("version,v", "");
+	// clang-format on
+
+	po::positional_options_description pod;
+	pod.add("config-file", 1);
+
+	set_ips_display_name("irods_s3_api");
+
+	try {
+		po::variables_map vm;
+		po::store(po::command_line_parser(_argc, _argv).options(opts_desc).positional(pod).run(), vm);
+		po::notify(vm);
+
+		if (vm.count("help") > 0) {
+			print_usage();
+			return 0;
+		}
+
+		if (vm.count("version") > 0) {
+			print_version_info();
+			return 0;
+		}
+
+		if (vm.count("dump-config-template") > 0) {
+			print_configuration_template();
+			return 0;
+		}
+
+		if (vm.count("dump-default-jsonschema") > 0) {
+			fmt::print(default_jsonschema());
+			return 0;
+		}
+
+		if (vm.count("config-file") == 0) {
+			fmt::print(stderr, "Error: Missing [CONFIG_FILE_PATH] parameter.");
+			return 1;
+		}
+
+		const auto config = json::parse(std::ifstream{vm["config-file"].as<std::string>()});
+		irods::http::globals::set_configuration(config);
+
+#if 0
+		{
+			const auto schema_file = (vm.count("jsonschema-file") > 0) ? vm["jsonschema-file"].as<std::string>() : "";
+			if (!is_valid_configuration(schema_file, vm["config-file"].as<std::string>())) {
+				return 1;
+			}
+		}
+#endif
```

Comment:
?

Reply:
That's the schema validation logic. I temporarily disabled it so it doesn't get in the way.

We can write up issues for revisiting bits of code like that.

Reply:
gotcha.